File: src/test/java/io/binac/leetcode/SingleNumberIITest.java
package io.binac.leetcode;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class SingleNumberIITest {
    private final SingleNumberII.Solution1 solution1 = new SingleNumberII.Solution1();

    @Test
    void test1() {
        int nums1[] = {2, 2, 3, 2};
        assertEquals(3, solution1.singleNumber(nums1));

        int nums2[] = {0, 1, 0, 1, 0, 1, 99};
        assertEquals(99, solution1.singleNumber(nums2));
    }
}

File: cpp/tests/SingleNumberIITest.cpp
#include "SingleNumberII.h"
#include "gtest/gtest.h"

using namespace lcpp;

TEST(SingleNumberII, Solution137_1) {
  auto S137_1 = Solution137_1();

  std::vector<int> Nums1{2, 2, 3, 2};
  int const Expected1 = 3;
  EXPECT_EQ(Expected1, S137_1.singleNumber(Nums1));

  std::vector<int> Nums2{0, 1, 0, 1, 0, 1, 99};
  int const Expected2 = 99;
  EXPECT_EQ(Expected2, S137_1.singleNumber(Nums2));
}